misk-jooq/src/main/kotlin/misk/jooq/JooqTransacter.kt

```diff
@@ -20,8 +20,12 @@ import org.jooq.impl.DataSourceConnectionProvider
 import org.jooq.impl.DefaultExecuteListenerProvider
 import org.jooq.impl.DefaultTransactionProvider
 import misk.logging.getLogger
+import java.sql.SQLException
+import java.sql.SQLRecoverableException
+import java.sql.SQLTransientException
 import java.time.Clock
 import java.time.Duration
+import javax.persistence.OptimisticLockException
```

Comment:
We talked offline that there may be a better jooq-specific exception, see: https://www.jooq.org/doc/latest/manual/sql-execution/crud-with-updatablerecords/optimistic-locking/